start upgrade to dotnet 8 min version, netstandard 2.0 for SDK

rvowles opened the pull request.

rvowles committed:
start upgrade to donet 6 min version

LavaToaster commented:
@rvowles Seeing as you've started doing a bit of an upgrade here, would it be possible for you to have a quick look at upgrading RestSharp, and getting this PR in? (or I can submit a PR for upgrading this separately, let me know if you'd like me to, and sorry if this is the wrong channel!)

Would help as I'm getting the same issue as #28. There seems to be a breaking change in the version desired by our service and the one desired by the featurehub sdk.

rvowles commented:
I'll need to do a major revision release, RestSharp doesn't seem to understand versioning unfortunately.

rvowles committed:
updated sdk, tests and web example

rvowles committed:
fixed tests and config

rvowles committed:
update

- make appropriate Dockerfile
for e2e testing with ToDoWeb and
confirmed it is passing tests
- update publishing to match examples

rvowles committed:
fix tag name usage in publish pipeline
